corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -125,7 +128,16 @@ hqDefine("data_dictionary/js/data_dictionary", [
         };
 
         self.deprecateProperty = function () {
-            self.deprecated(true);
+            if (!toggles.toggleEnabled('GEOSPATIAL') || !self.isGeoCaseProp()) {
+                self.deprecated(true);
+                return;
+            }
+
+            const $modal = $("#deprecate-geospatial-prop-modal").modal('show');
+            $("#deprecate-geospatial-prop-btn").on('click', function () {
+                self.deprecated(true);
+                $modal.modal('hide');
+            });
```

Comment:
I agree the opposite is a bit easier to read through. I'll update this.

Regarding the multiple bindings, since `isGeoCaseProp` can only be true for a single case property this binding will only ever be applied to one case property. But it might fire multiple times for the same case property, so I'll update to unbind the click before binding again. Nice catch!